src/apply.jl

```diff
@@ -339,17 +339,17 @@ function ITensors.apply(
     replaceind!(S, ind_to_replace, ind_to_replace_with')
     replaceind!(V, ind_to_replace, ind_to_replace_with)
 
-    ψv1, bond_tensors[e], ψv2 = U * Qᵥ₁, S, V * Qᵥ₂
+    ψv1, bond_tensors[e], bond_tensors[reverse(e)], ψv2 = U * Qᵥ₁, S, S, V * Qᵥ₂
```

Comment:
Also a bit unfortunate it got more complicated, this kind of code indicates it may be better to use a more sophisticated data structure like `DataGraph`.